src/main/scala/com/chaordicsystems/cactus/Parser.scala

```diff
@@ -0,0 +1,140 @@
+    val args = values("args")
+    Cactus(op, field, args)
+  }
+
+  def comparativeQueryHandler(op: Operation, property: String, value: Any): QueryDefinition = {
+    op match {
+      case LT => rangeQuery(property) to value includeUpper false
+      case GT => rangeQuery(property) from value includeLower false
+      case LE => rangeQuery(property) to value includeUpper true
+      case GE => rangeQuery(property) from value includeUpper true
+      case _  => matchQuery(property, value)
+    }
+  }
+
+  def comparativeQuery(op: Operation, property: String, value: Any): QueryDefinition = {
+    value match {
+      case _: Boolean => termQuery(property, value)
+      case _: Double  => comparativeQueryHandler(op, property, value.asInstanceOf[Double])
+      case _: BigInt  => comparativeQueryHandler(op, property, value.asInstanceOf[BigInt].toInt)
+      case _: String  => comparativeQueryHandler(op, property, value.asInstanceOf[String])
+      case _          => throw InvalidCactusQueryFormatException()
+    }
+  }
+
+  def handleDetails(cactus: Cactus): NestedQueryDefinition = {
+    val name = cactus.field.get.split("\\.")(1)
+    val valueType = cactus.args match {
+      case _: BigInt  => "value_int"
+      case _: Boolean => "value_bool"
+      case _: Double  => "value_float"
+      case _: String  => "value_str"
+      case _          => throw InvalidCactusQueryFormatException()
+    }
+
+    Operation.withName(cactus.op) match {
+      case NE => nestedQuery("details").query {
+        bool {
+          must(matchQuery("details.name", name))
+          not(comparativeQuery(NE, s"details.$valueType", cactus.args))
+        }
+      }
+      case op => nestedQuery("details").query {
+        bool {
+          must (
+            matchQuery("details.name", name),
+            comparativeQuery(op, s"details.$valueType", cactus.args)
+          )
+        }
+      }
+    }
+  }
+
+  def handleCatOrTag(cactus: Cactus): QueryDefinition = {
+    val field = if (cactus.field.isDefined) cactus.field.get else throw InvalidCactusQueryFormatException()
+    val listArgs = cactus.args.asInstanceOf[List[Any]]
+
+    Operation.withName(cactus.op) match {
+      case ALL =>
+        if (listArgs.length == 1) {
+          nestedQuery(field).query {
+            bool {
+              must(matchQuery(s"$field.id", listArgs.head))
+            }
+          }
+        } else {
+          bool {
+            must(
+              listArgs.map(x => {
+                nestedQuery(field).query {
+                  bool {
+                    must(matchQuery(s"$field.id", x))
+                  }
+                }
+              })
+            )
+          }
+        }
+
+      case ANY => nestedQuery(field).query {
+        bool {
+          should(listArgs.map(x => matchQuery(s"$field.id", x))) minimumShouldMatch 1
+        }
+      }
+    }
+  }
+
+  def validateAndTranslate(cactus: Cactus): QueryDefinition = {
```

Comment:
Too complicated code. High cohesion.
This god class is not good at all.
Imagine how cumbersome it will be to add or change an operator. Code is mixed among several methods, validations and operators.